Modules/_ctypes/ctypes.h

```diff
@@ -390,6 +395,7 @@ typedef struct {
     PyObject *checker;
     PyObject *module;
     int flags;                  /* calling convention and such */
+    int dict_final;
```

Comment:
Hi,
Don't have much time for a full review, all that comes to mind is conserving a bit of memory.

Could this be a bool or char, or would that hurt performance?

AFAIK flags can be uint16. And `mutex` can be moved here to fill the padding. And the `initialized` field might want to become a flag.
The rearranging could go in a separate PR, but I'd appreciate if you make `dict_final` smaller *here*, so I don't need to consult that change with free-threading experts :)

Reply:
I changed it to `uint8_t` type